mindmeld/models/nn_utils/classification.py

```diff
@@ -0,0 +1,490 @@
+        logger.info(msg)
+        return msg
+
+    def _to_device(self, batch_data: BatchData) -> BatchData:
+        """method for gpu usage and data loading"""
+        for k, v in batch_data.items():
+            if v is not None and isinstance(v, torch.Tensor):
+                batch_data[k] = v.to(self.params.device)
+            elif isinstance(v, list):
+                batch_data[k] = [
+                    vv.to(self.params.device) if isinstance(vv, torch.Tensor) else vv
+                    for vv in v
+                ]
+            elif isinstance(v, dict):
+                batch_data[k] = self._to_device(batch_data[k])
+        return batch_data
+
+    def fit(self, examples, labels, **params):  # pylint: disable=too-many-locals
+        """
+        Trains the underlying neural model on the inputted data and finally retains the best scored
+        model among all iterations.
+
+        Because of possibly large sized neural models, instead of retaining a copy of best set of
+        model weights on RAM, it is advisable to dump them in a temporary folder and upon training,
+        load the best checkpointed weights.
+        """
+
+        if self.ready:
+            msg = "The model is already fitted or is loaded from a file. Aborting re-fitting."
+            logger.error(msg)
+
+        # update params upon preparing encoder and embedder
+        params = {**self.get_default_params(), **params}
+        params = self._prepare_input_encoder(examples, **params)
+        params = self._prepare_embedder(**params)
+
+        # update params upon identifying unique labels
+        try:
+            num_labels = len(set(labels))
+        except TypeError:  # raised in cased on token classification
+            num_labels = len(set(sum(labels, [])))
+        params.update({"num_labels": num_labels})
+
+        # update params with label pad idx
+        try:  # used by token classifiers in crf masks and for training/evaluation
+            self.label_padding_idx = params["label_padding_idx"]
+            params.update({"label_padding_idx": params["label_padding_idx"]})
+        except KeyError:  # raised in cased on sequence classification
+            pass
+
+        # use all default params and inputted params to update self.params
+        self.params.update(params)
+
+        # init the graph and move model to device, inputs are moved to device on-the-go
+        self._init_graph()
+        self.to(self.params.device)
+        self.who_am_i()
+
+        # dumping weights during training process into a temp folder instead of keeping in
+        # memory to reduce memory usage
+        temp_folder = os.path.join(USER_CONFIG_DIR, "tmp", "pytorch_models", str(uuid.uuid4()))
+        temp_weights_save_path = os.path.join(temp_folder, "pytorch_model.bin")
+        os.makedirs(temp_folder, exist_ok=True)
+
+        # split input data into train & dev splits, and get data loaders
+        indices = np.arange(len(examples))
```

Comment:
Could we use this https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html